.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -0,0 +1,494 @@
+          until docker manifest inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/$image:${{ steps.vars.outputs.sha }} &> /dev/null; do sleep 45s; done
+          done
+
+      - name: Install Cilium ${{ env.cilium_stable_version }}
+        uses: cilium/little-vm-helper@908ab1ff8a596a03cd5221a1f8602dc44c3f906d # v0.0.12
+        with:
+          provision: 'false'
+          cmd: |
+            cd /host/
+
+            CILIUM_CLI_MODE=helm ./cilium-cli install \
+              ${{ steps.cilium-stable-config.outputs.config }}
+
+            ./cilium-cli status --wait
+            kubectl get pods --all-namespaces -o wide
+            kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+      - name: Start conn-disrupt-test
+        uses: cilium/little-vm-helper@908ab1ff8a596a03cd5221a1f8602dc44c3f906d # v0.0.12
+        with:
+          provision: 'false'
+          cmd: |
+            cd /host/
+
+            # Create pods which establish long lived connections. It will be used by
+            # subsequent connectivity tests with --include-conn-disrupt-test to catch any
+            # interruption in such flows.
+            ./cilium-cli connectivity test --include-conn-disrupt-test --conn-disrupt-test-setup
+
+      - name: Upgrade Cilium
+        uses: cilium/little-vm-helper@908ab1ff8a596a03cd5221a1f8602dc44c3f906d # v0.0.12
+        with:
+          provision: 'false'
+          cmd: |
+            cd /host/
+
+            CILIUM_CLI_MODE=helm ./cilium-cli upgrade \
+              ${{ steps.cilium-newest-config.outputs.config }}
+
+            ./cilium-cli status --wait
+            kubectl get pods --all-namespaces -o wide
+            kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+      - name: Test Cilium after upgrade
+        uses: cilium/little-vm-helper@908ab1ff8a596a03cd5221a1f8602dc44c3f906d # v0.0.12
+        with:
+          provision: 'false'
+          cmd: |
+            cd /host/
+
+            EXTRA=""
+            if [ "${{ matrix.secondary-network }}" = "true" ]; then
+              EXTRA="--secondary-network-iface=eth1"
+            fi
+
+            # Disable check-log-errors due to https://github.com/cilium/cilium-cli/issues/1858
+            ./cilium-cli connectivity test --include-unsafe-tests --collect-sysdump-on-failure \
+              --include-conn-disrupt-test \
+              --flush-ct \
+              --sysdump-hubble-flows-count=1000000 --sysdump-hubble-flows-timeout=5m \
+              --sysdump-output-filename "cilium-sysdump-${{ matrix.name }}-<ts>" \
+              --junit-file "cilium-junits/${{ env.job_name }} (${{ join(matrix.*, ', ') }}).xml" \
+              --junit-property github_job_step="Run tests upgrade 2 (${{ join(matrix.*, ', ') }})" \
+              \$EXTRA
+
+            # --flush-ct interrupts the flows, so we need to set up again.
+            ./cilium-cli connectivity test --include-conn-disrupt-test --conn-disrupt-test-setup
```

Comment:
Why is the conntrack table being flushed in the first place?

Reply:
It's because of the L7 connectivity tests, which we run before the conn-disrupt-tests. They can make subsequent tests flaky - https://github.com/cilium/cilium-cli/commit/6d4434fb60.